### ISSUE TITLE
shaders: Raspberry PI4 compatibility

### ISSUE THREAD

**Issue by cmitu** [residualvm/residualvm#1588]
Hello,

 I'm trying to run ResidualVM with the _grim-demo-win_ downloaded from your site using the `opengl_shaders` renderer, but loading a shader throws a GLSL compilation error. 

Here's the full log from running the demo:
```
Debuglevel (from command line): 2
Using SDL Video Driver "KMSDRM"
Using SDL Audio Driver "alsa"
Output sample rate: 44100 Hz
Output buffer size: 1024 samples
INFO: OpenGL Vendor: Broadcom
INFO: OpenGL Renderer: V3D 4.2
INFO: OpenGL Version: OpenGL ES 3.0 Mesa 19.2.0-rc1
INFO: OpenGL Red bits: 8
INFO: OpenGL Green bits: 8
INFO: OpenGL Blue bits: 8
INFO: OpenGL Alpha bits: 8
INFO: OpenGL Z buffer depth bits: 24
INFO: OpenGL Double Buffer: 0
INFO: OpenGL Stencil buffer bits: 8
INFO: GLSL version: OpenGL ES GLSL ES 3.00
Virtual keyboard pack 'vkeybd_default' loaded successfully
WARNING: No hardware input were defined, using defaults!
User picked target 'grim-demo-win' (engine ID 'grim', game ID 'grim')...
   Looking for a plugin supporting this target... Grim
Found game: grim (Demo Windows/English) (7)
 ... new best match, removing all previous candidates
Running Grim Fandango (Demo/Windows/English)
INFO: OpenGL Vendor: Broadcom
INFO: OpenGL Renderer: V3D 4.2
INFO: OpenGL Version: OpenGL ES 3.0 Mesa 19.2.0-rc1
INFO: OpenGL Red bits: 8
INFO: OpenGL Green bits: 8
INFO: OpenGL Blue bits: 8
INFO: OpenGL Alpha bits: 8
INFO: OpenGL Z buffer depth bits: 24
INFO: OpenGL Double Buffer: 0
INFO: OpenGL Stencil buffer bits: 8
INFO: GLSL version: OpenGL ES GLSL ES 3.00
Could not compile shader grim_background.vertex: 0:3(15): error: A shader cannot redefine built-in function `round' in GLSL ES 1.00
```

The PI4 has OpenGL ES 3.0 (for now) drivers and SDL2 was compiled with support for GLES, while ResidualVM was compiled with `--enable-opengl-shaders` and `--force-opengles2` (among othe options). The game runs fine with the `software` renderer, so it doesn't seem to be a game installation issue.

**Comment by Botje** [comment]
Can you try with the branch mentioned in #1584 ?

**Comment by cmitu** [comment]
Unfortunately, I get the same error as previously reported.

**Comment by Botje** [comment]
Try editing `opengl/shaders/compat_shaders.cpp` and removing the [two definitions of `round`](https://github.com/residualvm/residualvm/blob/master/graphics/opengl/compat_shaders.cpp#L30-L44) there.

**Comment by cmitu** [comment]
Yes, removing the the `round`'s definition from the shader make the game start and work. Thanks for the suggestion.

**Comment by Botje** [comment]
I tried a fix in [my `bugfix/es3-shaders` branch](https://github.com/Botje/residualvm/tree/edad0b80ce385c563c5e21ab2b9fb154b57c9e4d). Can you try with that as well? If that works I can push it to master.

**Comment by cmitu** [comment]
It doesn't work - it's still giving the same error, although the line number is a bit different now
>[...]
Could not compile shader grim_background.vertex: 0:4(15): error: A shader cannot redefine built-in function `round' in GLSL ES 1.00
!

The error comes from the MESA GLSL shader parser ([here](https://github.com/Igalia/mesa/blob/d6db858771ab72f607e43cd2cacf73b2f62fad30/src/compiler/glsl/ast_to_hir.cpp#L6074)), when the GLSL is 1.00, so I'm thinking that - despite the version GLSL version string number - the GLSL version in-driver is 1.00.

I tried forcing SDL to use a GLES3 context via [SDL_GL_CONTEXT_MAJOR_VERSION](https://github.com/residualvm/residualvm/blob/d538a8679f8fa55e6d41aa27160fb4b38f216c3c/backends/graphics/openglsdl/openglsdl-graphics.cpp#L365), but it doesn't seem to make a difference.

**Comment by Botje** [comment]
It's not a bug. The implementation is according to the specification where it was not before.

Can you try with https://github.com/Botje/residualvm/tree/00012b4dd8bb62146bc93b1e413f11c5bba0272c ? That sidesteps the problem by defining a ROUND macro instead.

**Comment by cmitu** [comment]
The latest commit seems to also get rid of the error, thank you.

**Comment by Botje** [comment]
Please do, if you have the time. I do not have access to any Pi at all :)

**Comment by cmitu** [comment]
Seems to work fine, loading the demo:
```
Debuglevel (from command line): 2
Using SDL Video Driver "RPI"
Using game controller: iBuffalo SNES Controller
Using SDL Audio Driver "alsa"
Output sample rate: 44100 Hz
Output buffer size: 1024 samples
INFO: OpenGL Vendor: Broadcom
INFO: OpenGL Renderer: VideoCore IV HW
INFO: OpenGL Version: OpenGL ES 2.0
INFO: OpenGL Red bits: 8
INFO: OpenGL Green bits: 8
INFO: OpenGL Blue bits: 8
INFO: OpenGL Alpha bits: 8
INFO: OpenGL Z buffer depth bits: 24
INFO: OpenGL Double Buffer: 1
INFO: OpenGL Stencil buffer bits: 8
INFO: GLSL version: OpenGL ES GLSL ES 1.00
Virtual keyboard pack 'vkeybd_default' loaded successfully
WARNING: No hardware input were defined, using defaults!
User picked target 'grim-demo-win' (engine ID 'grim', game ID 'grim')...
   Looking for a plugin supporting this target... Grim
Found game: grim (Demo Windows/English) (7)
 ... new best match, removing all previous candidates
Running Grim Fandango (Demo/Windows/English)
INFO: OpenGL Vendor: Broadcom
INFO: OpenGL Renderer: VideoCore IV HW
INFO: OpenGL Version: OpenGL ES 2.0
INFO: OpenGL Red bits: 8
INFO: OpenGL Green bits: 8
INFO: OpenGL Blue bits: 8
INFO: OpenGL Alpha bits: 8
INFO: OpenGL Z buffer depth bits: 24
INFO: OpenGL Double Buffer: 1
INFO: OpenGL Stencil buffer bits: 8
INFO: GLSL version: OpenGL ES GLSL ES 1.00
WARNING: !!!! Trying to call MakeSectorActive without a scene!
WARNING: Could not find bitmap !
WARNING: Could not find bitmap <none>!
WARNING: Lua_V1::SetGamma, implement opcode, level: 0.000000!
Using game controller: iBuffalo SNES Controller
```

**Comment by Botje** [comment]
Fixed in dbe9763af3108d442a21f0400748f7bb14cb8fa8.

Thanks for reporting the issue and aiding with debugging!
